eng/common/scripts/job-matrix/README.md

```diff
@@ -402,49 +488,57 @@ The top level keys are used as job names, meaning they get displayed in the azur
 
 The logic for generating display names works like this:
 
-- Join parameter values by "_"
-    a. If the parameter value exists as a key in `displayNames` in the matrix config, replace it with that value.
-    b. For each name value, strip all non-alphanumeric characters (excluding "_").
+* Join parameter values by `_`  
+    a. If the parameter value exists as a key in [`displayNames`](#displaynames) in the matrix config, replace it with that value.  
+    b. For each name value, strip all non-alphanumeric characters (excluding "_").  
     c. If the name is greater than 100 characters, truncate it.
 
-#### Filters
+### Filters
 
-Filters can be passed to the matrix as an array of strings, each matching the format of `<key>=<regex>`. When a matrix entry
-does not contain the specified key, it will default to a value of empty string for regex parsing. This can be used to specify
-filters for keys that don't exist or keys that optionally exist and match a regex, as seen in the below example.
+Filters can be passed to the matrix as an array of strings, in `MatrixFilters` parameter, each matching the format of `<key>=<regex>`.
+When a matrix combination does not contain the specified key, it will default to a value of empty string for regex parsing.
 
-Display name filters can also be passed as a single regex string that runs against the [generated display name](#displaynames) of the matrix job.
-The intent of display name filters is to be defined primarily as a top level variable at template queue time in the azure pipelines UI.
+Interpreting missing key in given combination as a key with value being empty string enables following scenarios:
```

Comment:
I think the way this sentence is written is a little confusing.

Reply:
Thanks for pointing this out. Addressed! Please have a look: https://github.com/Azure/azure-sdk-tools/pull/5023/commits/13657fe5258bd51e6b9bff8c1975c4566b220086 :)